dashboard/app/models/survey_result.rb

```diff
@@ -28,11 +28,23 @@ class SurveyResult < ActiveRecord::Base
     "other" => "Other/Unknown"
   }.freeze
 
-  DIVERSITY_ATTRS = ETHNICITIES.keys.map {|key| "diversity_#{key}"}
-  DIVERSITY_ATTRS << "diversity_farm"
+  DIVERSITY_ATTRS = (ETHNICITIES.keys.map {|key| "diversity_#{key}"} + ['diversity_farm']).freeze
 
   NET_PROMOTER_SCORE_ATTRS = %w(nps_value nps_comment).freeze
 
+  FREE_RESPONSE_ATTRS = %w(nps_comment).freeze
+  NON_FREE_RESPONSE_ATTRS = %w(
+    diversity_white
+    diversity_black
+    diversity_hispanic
+    diversity_asian
+    diversity_asian
+    diversity_native
+    diversity_american_indian
+    diversity_other
+    diversity_farm
+    nps_value
+  ).freeze
```

Comment:
Note that there is no intention to use `NON_FREE_RESPONSE_ATTRS`, other than in the test. If we decide the test is not worth having (confirming developers explicitly specify whether an attribute is freeform or not), this can be removed.